drivers/platform/maxim/max32670/maxim_irq.c

```diff
@@ -0,0 +1,581 @@
+ *    contributors may be used to endorse or promote products derived
+ *    from this software without specific prior written permission.
+ *  - The use of this software may or may not infringe the patent rights
+ *    of one or more patent holders.  This license does not release you
+ *    from the requirement that you obtain separate licenses from these
+ *    patent holders to use this software.
+ *  - Use of the software either in source or binary form, must be run
+ *    on or directly connected to an Analog Devices Inc. component.
+ *
+ * THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT,
+ * MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+ * IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+
+/******************************************************************************/
+/************************* Include Files **************************************/
+/******************************************************************************/
+
+#include <stdint.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <errno.h>
+#include "rtc.h"
+#include "uart.h"
+#include "tmr.h"
+#include "maxim_irq.h"
+#include "maxim_rtc.h"
+#include "max32670.h"
+#include "no_os_irq.h"
+#include "no_os_list.h"
+#include "no_os_uart.h"
+#include "no_os_rtc.h"
+#include "no_os_util.h"
+
+static struct event_list _events[] = {
+	[NO_OS_EVT_GPIO] = {.event = NO_OS_EVT_GPIO},
+	[NO_OS_EVT_UART_TX_COMPLETE] = {.event = NO_OS_EVT_UART_TX_COMPLETE},
+	[NO_OS_EVT_UART_RX_COMPLETE] = {.event = NO_OS_EVT_UART_RX_COMPLETE},
+	[NO_OS_EVT_UART_ERROR] = {.event = NO_OS_EVT_UART_ERROR},
+	[NO_OS_EVT_RTC] = {.event = NO_OS_EVT_RTC},
+	[NO_OS_EVT_TIM_ELAPSED] = {.event = NO_OS_EVT_TIM_ELAPSED},
+};
+
+extern mxc_uart_req_t uart_irq_state[MXC_UART_INSTANCES];
+extern bool is_callback;
+
+/******************************************************************************/
+/************************ Functions Definitions *******************************/
+/******************************************************************************/
+
+/**
+ * @brief Action comparator function
+ * @param data1 - List element
+ * @param data2 - Key
+ * @return 0 if the two are equal, any other integer otherwise
+ */
+int32_t irq_action_cmp(void *data1, void *data2)
```

Comment:
use `int` for the return type of the functions. check the entire code for these occurrences.

Reply:
no_os_irq_platform_ops structure specifies return type as 'int32_t' for all functions. Do I need to change type of these function definitions to int, as compiler will through warnings?